sb-update

```diff
@@ -143,18 +143,22 @@ def main():
             logging.StreamHandler(),
         ],
     )
-    update_file = options.path / UPDATE_FILENAME
+    try:
+        update_file = options.path / UPDATE_FILENAME
 
-    logging.info("Starting updates from {}".format(update_file))
-    add_debs_to_local_repo_from(update_file)
+        logging.info("Starting updates from {}".format(update_file))
+        add_debs_to_local_repo_from(update_file)
 
-    update_file.unlink()
+        update_file.unlink()
 
-    rebuild_apt_repo()
-    update_and_upgrade()
+        rebuild_apt_repo()
+        update_and_upgrade()
 
-    logging.info("Upgrade complete, rebooting.")
-    subprocess.check_call(["reboot"])
+        logging.info("Upgrade complete, rebooting.")
+        subprocess.check_call(["reboot"])
+    except Exception:
+        logging.exception("Upgrade failed")
```

Comment:
From what i'm seeing, this doesn't actually log the exception, just that one occurred. Perhaps we can stringify the exception and put that in the logs too?

Reply:
[`logging.exception`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) is clever and pulls the exception info from the `except` block for you (via [`sys.exc_info()`](https://docs.python.org/3/library/sys.html#sys.exc_info)), including a stack trace:
```python
>>> import logging
>>> try:
...  raise Exception("bees")
... except Exception:
...  logging.exception("fail!")
...  print('after')
... 
ERROR:root:fail!
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
Exception: bees
after
>>>
```